### ISSUE TITLE
Zipping logs on period basis

### ISSUE THREAD

**Issue by daneel3001** [NLog/NLog#3711]
Hello

I am trying to find the right combination of settings for **FileAsyncTarget** which will allow me to archive logs in this way:
- Every Sunday (if weekly archive file), archive existing logs, via **archiveEvery**
- Keep log files to around 500KB each, via **archiveAboveSize**
- Keep as many as 50 weekly archives, via **maxArchiveFiles**
- Archive logs to be zipped using archive file name layout of "archive-{####}.zip", via **archiveFileName**

Today I am using the following configuration but I don't think it's quite working and it might be due to conflicting settings, specifically **archiceAboveSize** vs **archiveEvery** properties.

```
<target xsi:type="File" encoding="utf-8"
                fileName="${logDirectory}/${fileBaseName}.log"				
                archiveFileName="${logDirectory}/${fileBaseName}.{####}.zip"                
                archiveAboveSize="512000" 
                maxArchiveFiles="50"				
                archiveEvery="Sunday"
                concurrentWrites="false"
				enableArchiveFileCompression="true"
				archiveOldFileOnStartup="false"
                layout="${layout}" />
      </target>
```

Many thanks in advance

**Comment by snakefoot** [comment]
NLog has no such functionality. You need to setup a scheduled-task or cron-job to make that zipping-logic happen.

**Comment by snakefoot** [comment]
You are ofcourse welcome to create a pullrequest that resolves #1249 and #2732. Then the file-compression could be extended to perform bulk compression in background.

**Comment by daneel3001** [comment]
Hello

Ok thanks for letting me know these aren't available at the moment.
I think we will consider the script route as suggested.
